google/cloud/spanner_dbapi/connection.py

```diff
@@ -175,43 +178,58 @@ def _rerun_previous_statements(self):
         from the last transaction.
         """
         for statement in self._statements:
-            res_iter, retried_checksum = self.run_statement(statement, retried=True)
-            # executing all the completed statements
-            if statement != self._statements[-1]:
-                for res in res_iter:
-                    retried_checksum.consume_result(res)
-
-                _compare_checksums(statement.checksum, retried_checksum)
-            # executing the failed statement
+            if isinstance(statement, list):
+                statements, checksum = statement
+
+                transaction = self.transaction_checkout()
+                status, res = transaction.batch_update(statements)
+
+                if status.code == ABORTED:
+                    self.connection._transaction = None
+                    raise Aborted(status.details)
+                elif status.code != OK:
+                    raise OperationalError(status.details)
```

Comment:
This should not be propagated to the client application. The status code should be included in the checksum check, and if the statement returned the same status code as the during the initial attempt, then everything is fine.